### ISSUE TITLE
remoteproc_mmap returns address which can be 0

### ISSUE THREAD

**Issue by bentheredonethat** [OpenAMP/open-amp#236]
1 solution is to move returned address to be an argument and make return value success/failure-reason

**Comment by arnopo** [comment]
Do you have a use case that highlights the fact that 0 can be a valid virtual address?
If 0 can be valid, yes, this API needs to be updated, if not, returning NULL in case of failure seems reasonable to me.

**Comment by bentheredonethat** [comment]
yes address can be 0 on Xilinx Zynq UltraScale Plus on RPU running baremetal or FreeRTOS application. The address 0 relates to the lowest value on TCM bank 0A for RPU0 and TCM bank 1A for RPU1.

**Comment by arnopo** [comment]
what about returning METAL_BAD_VA?

**Comment by edmooring** [comment]
That is probably the best solution, since it doesn't change the documented API. However, the comments in openamp/remoteproc.h need to be enhanced to show the error return value, and any examples that depend on the previous behavior should also be updated. It should probably also be highlighted in the release notes.
